tss-esapi/src/abstraction/ak.rs

```diff
@@ -228,8 +230,36 @@ pub fn load_ak(
     Ok(key_handle)
 }
 
-/// This creates an Attestation Key in the Endorsement hierarchy
+/// This creates an Attestation Key in the Endorsement hierarchy.
 pub fn create_ak<IKC: IntoKeyCustomization>(
```

Comment:
I think it'd be good to mark one of them as [`#[deprecated]`](https://doc.rust-lang.org/reference/attributes/diagnostics.html#the-deprecated-attribute)...

Reply:
I was thinking about that. The problem is that the function will not be removed but it will have its arguments changed in the next major version.

So xx will have its arguments changed.
and xx_2 will be removed. So xx_2 is the function that we want people to use right now. But it will be removed so marking it deprecated here could cause a lot of confusion.

Reply:
Good point, so let's not do that. But a doc-comment WARNING note or something may still be a good idea? :thinking:

Reply:
That is an excellent idea. I need to rephrase the comments any way because I could barely understand them my self when I tried to look at the code in the browser.

Reply:
I think I have managed to fix it now. Please have a last look at the comments so I have not made some embarrassing mistakes @wiktor-k @ionut-arm

Reply:
I think it looks really nice :ok_hand: